style_editor.html

```diff
@@ -9391,87 +9403,99 @@
+          title="Press this to trigger the EFFECT selected in the More Effects menu.">
+      </span>
     
-<script>
-  // Get the menu and button elements
-  const menu = document.getElementById('more_effects_menu');
-  const button = document.getElementById('do-selected');
-
-  // Create a dictionary to store actions for each item
-  const actions = {};
-
-  // Add values from the enum builder to an array and sort alphabetically
-  const values = Object.entries(EFFECT_ENUM_BUILDER.value_to_name).sort((a, b) => a[1].localeCompare(b[1]));
-
-  // Add sub-groups for the different categories of effects
-  const generalEffectsMenu = document.createElement('optgroup');
-  generalEffectsMenu.label = 'General Effects';
-  const userEffectsMenu = document.createElement('optgroup');
-  userEffectsMenu.label = 'User Effects';
-  const blasterEffectsMenu = document.createElement('optgroup');
-  blasterEffectsMenu.label = 'Blaster Effects';
-  const gameEffectsMenu = document.createElement('optgroup');
-  gameEffectsMenu.label = 'Game Effects';
-
-  // Add sorted values to the menu and actions dictionary, 
-  // excluding effects with dedicated buttons and error msgs.
-  for (const [value, name] of values) {
-    if (![
-      EFFECT_CLASH,
-      EFFECT_STAB,
-      EFFECT_BLAST,
-      EFFECT_FORCE,
-      EFFECT_BOOT,
-      EFFECT_NEWFONT,
-      EFFECT_PREON,
-      EFFECT_ERROR_IN_BLADE_ARRAY,
-      EFFECT_ERROR_IN_FONT_DIRECTORY,
-      EFFECT_FONT_DIRECTORY_NOT_FOUND,
-      EFFECT_SD_CARD_NOT_FOUND
-    ].includes(Number(value))) {
-      const option = document.createElement('option');
-      option.value = value;
-      option.text = name;
-
-      // Check if the effect belongs to a certain category and add it to the corresponding sub-group
-      if (name.startsWith('EFFECT_GAME')) {
-        gameEffectsMenu.appendChild(option);
-      } else if (name.startsWith('EFFECT_USER')) {
-        userEffectsMenu.appendChild(option);
-      } else if ([
-          EFFECT_STUN, 
-          EFFECT_FIRE, 
-          EFFECT_CLIP_IN, 
-          EFFECT_CLIP_OUT, 
-          EFFECT_RELOAD, 
-          EFFECT_MODE, 
-          EFFECT_RANGE, 
-          EFFECT_EMPTY, 
-          EFFECT_FULL, 
-          EFFECT_JAM, 
-          EFFECT_UNJAM, 
-          EFFECT_PLI_ON, 
-          EFFECT_PLI_OFF
-        ].includes(Number(value))) {
-        blasterEffectsMenu.appendChild(option);
-      } else {
-        generalEffectsMenu.appendChild(option);
-      }
-      actions[value] = function() { blade.addEffect(value, 0.0) };
-    }
-  }
+  <script>
+         // Get the menu and button elements
+        const menu = FIND('more_effects_menu');
+        const button = FIND('do_selected');
+
+        // Create an object to store actions for each item
+        const actions = {};
+
+        /* Add values from the enum builder to an array and sort alphabetically,
+        excluding effects with dedicated buttons.*/
```

Comment:
I think we should consider removing some of those dedicated buttons, but that can be done later...